Zotero/Scenes/Detail/PDF/PDFCoordinator.swift

```diff
@@ -199,10 +199,31 @@ extension PDFCoordinator: PdfReaderCoordinatorDelegate {
         let viewController = PDFSearchViewController(controller: pdfController, text: text)
         viewController.delegate = delegate
         viewController.overrideUserInterfaceStyle = userInterfaceStyle
-        viewController.modalPresentationStyle = UIDevice.current.userInterfaceIdiom == .pad ? .popover : .formSheet
-        viewController.popoverPresentationController?.barButtonItem = sender
+        setupPresentation(for: viewController, with: sender)
         self.pdfSearchController = viewController
         self.navigationController?.present(viewController, animated: true, completion: nil)
+
+        func setupPresentation(for pdfSearchController: PDFSearchViewController, with sender: UIBarButtonItem) {
+            pdfSearchController.modalPresentationStyle = UIDevice.current.userInterfaceIdiom == .pad ? .popover : .formSheet
+            guard let popoverPresentationController = pdfSearchController.popoverPresentationController else { return }
+            if navigationController?.isNavigationBarHidden == false {
+                if #available(iOS 17, *) {
+                    popoverPresentationController.sourceItem = sender
+                } else {
+                    popoverPresentationController.barButtonItem = sender
+                }
+                popoverPresentationController.sourceView = nil
+                popoverPresentationController.sourceRect = .null
```

Comment:
Why use `.null` here and `.zero` below? Is there a difference for this case?

Reply:
Yes, `.null` is the default value, which instructs the system to use the current frame of `sourceView`, that I think would put the popover outside of the visible screen.
`.zero` anchors it on the top left corner of the screen, which is what was already happening if the search popover was triggered by key press, when the navigation bar is hidden (but has been properly setup first).

Reply:
On the other hand, we can probably always set it to `.zero`, since it will be ignored if `sourceView` is `nil` anyway.

Reply:
Sure you can change it but this should be fine too.

Reply:
Check https://github.com/zotero/zotero-ios/pull/823/commits/d9abe2729c398b5bdf3a87fb770858f76ed97616